### PR TITLE
Change when release_version to be more readable

### PR DESCRIPTION
**Pull request by reihl**
I replaced the place I found with ansible.builtin.version

Test-Hints: no-check

### CODE REVIEW

**Comment by manurodriguez** [comment]
recheck

File: roles/node_prep/tasks/10_validation.yml [hunk 1]
```diff
@@ -348,7 +348,9 @@
 - name: Fail if dual-stack is requested and build does not support it
   fail:
     msg: This release {{ release_version }} does not support dual-stack deployments
-  when: release_version.split('.')[0]|int == 4 and release_version.split('.')[1]|int < 6 and dualstack_baremetal
+  when:
+    - release_version is ansible.builtin.version('4.6', '<')
+    - dualstack_baremetal
```
Comment (manurodriguez) [inline roles/node_prep/tasks/10_validation.yml:353]: now that you are making easier to read these conditions, can we be more explicit what are we checking with dualstack_baremetal? Maybe: `dualstack_baremetal | bool`, wdyt?

File: roles/node_prep/tasks/40_bridge.yml [hunk 2]
```diff
@@ -67,7 +67,7 @@
         state: present
       when: (not enable_virtualmedia) and
             ((ipv4_provisioning|bool) or (not ipv6_enabled|bool) or
-            ((release_version.split('.')[0]|int == 4) and (release_version.split('.')[1]|int == 3)))
+            (release_version is ansible.builtin.version('4.3', '=='))
```
Comment (manurodriguez) [inline roles/node_prep/tasks/40_bridge.yml:70]: I think here we are missing a `)` to close the or expression that started in the line above. 
```suggestion
            (release_version is ansible.builtin.version('4.3', '==')))
```
https://www.distributed-ci.io/jobs/07318323-753a-4add-b0ed-7295b7183cef/jobStates?sort=date&task=855b07b5-2f23-441f-b7ac-0e8f6615be48

File: roles/node_prep/tasks/40_bridge.yml [hunk 3]
```diff
@@ -80,7 +80,7 @@
...
```
Comment (manurodriguez) [inline roles/node_prep/tasks/40_bridge.yml:83]: same here, I believe we are missing to close a parenthesis at the end of this line, to group all the `or` conditions.
```suggestion
            (release_version is ansible.builtin.version('4.3', '==')))
```

File: roles/node_prep/tasks/45_networking_facts.yml [hunk 4]
```diff
@@ -37,7 +37,7 @@
     ip: "{{ ansible_facts[provisioning_bridge]['ipv4']['address'] }}/{{ ansible_facts[provisioning_bridge]['ipv4']['netmask'] }}"
   when: (not enable_virtualmedia) and
         ((ipv4_provisioning|bool) or (not ipv6_enabled|bool) or
-        ((release_version.split('.')[0]|int == 4) and (release_version.split('.')[1]|int == 3)))
+        (release_version is ansible.builtin.version('4.3', '=='))
```
Comment (manurodriguez) [inline roles/node_prep/tasks/45_networking_facts.yml:40]: An another parenthesis to close
```suggestion
        (release_version is ansible.builtin.version('4.3', '==')))
```

**Review by manurodriguez — CHANGES_REQUESTED** [review]
Thanks @reihl, just minors things to fix.

File: roles/node_prep/tasks/10_validation.yml (hunk 1 above)
Comment (reihl) [reply roles/node_prep/tasks/10_validation.yml]: 100% agree, done

File: roles/node_prep/tasks/40_bridge.yml (hunk 2 above)
Comment (reihl) [reply roles/node_prep/tasks/40_bridge.yml:70]: True, changed

File: roles/node_prep/tasks/40_bridge.yml (hunk 3 above)
Comment (reihl) [reply roles/node_prep/tasks/40_bridge.yml:83]: True, changed

File: roles/node_prep/tasks/45_networking_facts.yml (hunk 4 above)
Comment (reihl) [reply roles/node_prep/tasks/45_networking_facts.yml:40]: True, changed

**Review by ramperher — DISMISSED** [review]
LGTM

**Review by manurodriguez — APPROVED** [review]
Thanks!

File: roles/node_prep/tasks/40_bridge.yml [hunk 5]
```diff
@@ -96,8 +96,7 @@
       when:
         - not enable_virtualmedia|bool
         - ipv6_enabled|bool
-        - release_version.split('.')[0]|int == 4 and release_version.split('.')[1]|int > 3 or
-          release_version.split('.')[0]|int > 4
+        - release_version is ansible.builtin.version('4.3', '>=')
```
Comment (derekhiggins) [inline roles/node_prep/tasks/40_bridge.yml:99]: Looks like this one should be '>'

File: roles/node_prep/tasks/40_bridge.yml [hunk 6]
```diff
@@ -112,8 +111,7 @@
...
```
Comment (derekhiggins) [inline roles/node_prep/tasks/40_bridge.yml:114]: Looks like this one should be '>'

Comment (reihl) [reply roles/node_prep/tasks/40_bridge.yml]: Fix, thanks!!

**Review by ramperher — APPROVED** [review]
LGTM

**Review by tonyskapunk — APPROVED** [review]
:+1:  @reihl please squash your commits before merging.

https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/575709d779f30fea5154ae66c829aaa1ecaa93fc/docs/dev.md?plain=1#L31-L34

**Pull request merged**